### PR TITLE
Stop 20x Melatonin Spawns

### PR DESCRIPTION
**Pull request by TheMurderUnicorn**
#### Summary
Bugfixes "Stop 20x Melatonin Spawns"

#### Purpose of change

In the itemgroup "shelter" which is used in everything from zombie pockets to stores (this is a terrible group but I can't be asked to fix this huge mess), melatonin was being spawned 20 times.  IE: prob 20, but when that prob won, 20 bottles of 10 spawned instead of 1 bottle.

#### Describe the solution

Remove the offensive "count".  Also, other pills in this itemgroup spawn in partially depleted bottles whereas melatonin spawned in full bottles.  I have added a group to allow for the 10 pill bottle to spawn 1-10 pills instead, and hooked that in.

#### Describe alternatives you've considered

None really, clearly wasn't meant to spawn 20 bottles of pills in one selection.

#### Testing

Made sure the itemgroups worked, tested them, etc.

#### Additional context
Old spawns of the group:
image

New spawn example (showing a 5 spawn and a 10 spawn), which also illustrates the random number is working:
image

Oop and it shows vitamin C supplements, if anyone cares that was what I was working on when I found this bug, we should have vitamin C supplements soon provided I can muster the strength to endure nested itemgroups.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
Depleted bottles of stuff don't make sense in shops, but do make sense in homes (do people really carry around bottles of drugs in real life [apart from people who actually have to due to their dosage regimen, in which case they probably don't carry bottles, but dose dispensers or blister packs]?).

**Comment by TheMurderUnicorn** [comment]
So unfortunately, there is no distinction between that depending on the itemgroup.  This particular itemgroup had pills already spawning in is 'shelter' which hits everything from zombie pockets to a couple stores.  There is no way for me to easily address this.  If it makes you feel better, in the dominant store itemgroups (like vitamin shop, which hits pharmacies and mall stores) spawn them in full bottles.  This is one of the itemgroups that is overused and hits situations where they should be full and where they should be depleted.  I'm not sure what to do here, I could change the group to be full bottles but then I should also change all the other supplements/pills that appear in the group to be full bottles, which is also bad because it would spawn like a survivor zombie with full bottles of pills in their pocket or whatever.  

TLDR: I can't really defend the choice other than "The other pills here are like this", and I don't feel there's an objective correct answer so this is what I chose to do.

**Comment by PatrikLundell** [comment]
The objective right answer is to use one group for stores (and potential other places containing unused stuff, such as warehouses [not to be confused with places where people store their own junk]), and another for consumer loot.
I'm not saying you are have to fix the mess because you touched it, though. That would be scope creep.

**Comment by TheMurderUnicorn** [comment]
I meant there is no objective right answer on whether THIS group in THIS situation should contain partially depleted bottles of melatonin or full bottles.

Examples of where I found the shelter group:
Locations:
Church retreat, used alongside SUS clothing spawns
Backrooms mod - Random cardboard box contents (alongside dry goods, drugs, condiments, random crap)
Megastores
Refugee Center
LMOE


Mobs:
Survivor and elite survivor death drops
Feral survivalists
Feral Militia

With the exception of megastores, none of these locations/mobs NEED to spawn full bottles.  They could be expected to spawn partially depleted bottles.  So, if someone wants to knock out the megastore using the "shelter" group, this would be resolved.  I have no plans to do this, but based on this additional information, I think spawning them as partially depleted in this PR makes sense.
